emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -0,0 +1,161 @@
+            reading=elapsed_ms
+        )
+        print(f"Function '{function_name}' executed successfully in {elapsed_ms:.2f} ms.")
+        logging.info(f"Function '{function_name}' executed successfully in {elapsed_ms:.2f} ms.")
+
+        # Verification: Adjusted Query to Match Document Structure
+        timeseries_db = gdb.get_timeseries_db()
+        
+        # Define a time window (e.g., +/- 5 seconds) to account for any timing discrepancies
+        time_window = 5  # seconds
+        current_time = time.time()
+        query = {
+            "metadata.key": "stats/dashboard_time",
+            "data.name": function_name,
+            "data.ts": {"$gte": timestamp - time_window, "$lte": timestamp + time_window},
+            "data.reading": {"$gte": elapsed_ms - 10, "$lte": elapsed_ms + 10}  # 10 ms tolerance
+        }
+        
+        # Retrieve the most recent document for the function
+        stored_document = timeseries_db.find_one(
+            query,
+            sort=[("data.ts", -1)]
+        )
+
+        if stored_document:
+            # Inspect the stored document
+            stored_ts = stored_document.get("data", {}).get("ts", 0)
+            stored_reading = stored_document.get("data", {}).get("reading", 0)
+            logging.debug(f"Stored Document for '{function_name}': ts={stored_ts}, reading={stored_reading}")
+
+            # Check if the reading is within a reasonable tolerance (e.g., ±100 ms)
+            if abs(stored_reading - elapsed_ms) <= 100:
+                print(f"Verification passed: Data for '{function_name}' is stored correctly.")
+                logging.info(f"Verification passed: Data for '{function_name}' is stored correctly.")
+            else:
+                print(f"Verification failed: 'reading' value for '{function_name}' is outside the expected range.")
+                logging.error(f"Verification failed: 'reading' value for '{function_name}' is outside the expected range.")
+        else:
+            print(f"Verification failed: Data for '{function_name}' was not found in the database.")
+            logging.error(f"Verification failed: Data for '{function_name}' was not found in the database.")
+
+    except Exception as e:
+        # Even if the function fails, capture the elapsed time up to the exception
+        elapsed_seconds = timer.elapsed if 'timer' in locals() else 0  # Accessing the float attribute directly
+        elapsed_ms = elapsed_seconds * 1000
+
+        # Store the error timing
+        sdq.store_dashboard_error(
+            code_fragment_name=function_name,
+            ts=timestamp,
+            reading=elapsed_ms
+        )
+        print(f"Function '{function_name}' failed after {elapsed_ms:.2f} ms with error: {e}")
+        logging.error(f"Function '{function_name}' failed after {elapsed_ms:.2f} ms with error: {e}")
+
+        # Verification: Adjusted Error Query to Match Document Structure
+        timeseries_db = gdb.get_timeseries_db()
```

Comment:
please change to use the timeseries interface to read the data as well. That will make it easier to (potentially) switch to a different database under the hood later. if we choose to switch a specialized timeseries DB/event DB instead of mongodb.

Reply:
@shankari What do you mean by timeseries interface?

Reply:
@shankari this is the built in edb method, i just named it gdb, can rename it to edb like your previous functions

Reply:
is this what you meant? `timeseries_db = builtin_ts.get_timeseries_db(key="stats/dashboard_time")`

Reply:
No, I meant that you should use the timeseries interface (defined in `emission.storage.timeseries.abstract_timeseries`  - look at existing implementations in the server or dashboard code for examples.

Do *not* use the mongodb collections directly - the new methods will not work if we switch to a different underlying database later.

Reply:
Ah I was trying to find regular time series; did not realize it was abstract

Reply:
@shankari I used the `abstract_timeseries`; modified `_get_query` in the built-in to take a dict AND object as opposed to just object (for verification that we are putting the items in the db)

Reply:
also added more logging/error check for debugging